### ISSUE TITLE
Use the `keyboard_types` crate in winit

### ISSUE THREAD

**Issue by amrbashir** [rust-windowing/winit#2394]
Using the [`keyboard_types`](https://docs.rs/keyboard-types/latest/keyboard_types/) crate as a standardized keyboard crate among the ecosystem. see https://github.com/tauri-apps/tao/issues/460

**Comment by madsmtm** [comment]
Good idea! Will probably have to wait until we get the new keyboard API stuff worked out though: https://github.com/rust-windowing/winit/issues/1806

**Comment by madsmtm** [comment]
I've opened https://github.com/pyfisch/keyboard-types/issues/19 for tracking this in `keyboard-types`

**Comment by madsmtm** [comment]
This is also desired by Xilem / Masonry

**Comment by nicoburns** [comment]
This would also be helpful for us in https://github.com/DioxusLabs/blitz. Our use case is to be able to make crates that implement keyboard shortcuts (and thus need to be able to respond to key events) without depending on winit (and bringing in all the actual platform integration code). Some kind of `winit_types` or `winit_keyboard_types` crates that separately exports winit's existing types would also work for this use case, although an ecosystem-wide "vocabulary crate" would still be ideal.

**Comment by Exotik850** [comment]
The need to expose `keyboard_types` data from winit remains an important issue for the ecosystem. Implementing this through either a feature flag or a separate `winit_keyboard_types` crate (as discussed previously) would significantly reduce the current pain points many developers face. Right now, crates that need to work with winit's keyboard events are forced to either duplicate type definitions or write potentially error-prone conversion code. By making these types more accessible, we could reduce code duplication across projects and make it much simpler for crates to properly handle keyboard input without requiring the full winit dependency. This would be especially valuable for libraries that only need to process keyboard data *before* window management functionality. Based on previous discussions, both the feature flag and separate crate approaches seem viable - I'm curious which path the maintainers think would better serve the community's needs.

**Comment by HactarCE** [comment]
> might need some way to query the keyboard layout?

I have implemented this for Windows and Linux in my [`key-names`](https://crates.io/crates/key-names) crate (which would greatly benefit from a `winit-keyboard-types` crate or similar). My understanding is that this is possible on macOS (I know newer versions of Minecraft do it) and that Google Chrome has an API for it but other browsers probably do not.

**Comment by madsmtm** [comment]
I agree that the modifiers that `keyboard-types` expose is currently insufficient. My plan with https://github.com/rust-windowing/winit/pull/4026 is to use the key+code+location types, then we can consider improving the modifiers in `keyboard-types` later (ideally after we have settled on a good design in Winit).
